base/util.go

```diff
@@ -48,24 +48,41 @@ const (
 	kMaxDeltaTtlDuration = 60 * 60 * 24 * 30 * time.Second
 )
 
-// basicAuthURLRegexp is used to match the HTTP basic auth component of a URL
-var basicAuthURLRegexp = regexp.MustCompilePOSIX(`:\/\/[^:/]+:[^@/]+@`)
```

Comment:
Looking back at this regexp, it's not matching the full string, so would match strings containing URLs.
I can see at least one usage in SG where this behaviour is expected from the redact functions:

https://github.com/couchbase/sync_gateway/blob/9d12bf43928e56610e897e8eb836fe386343a846/db/sg_replicate_cfg.go#L195-L199

This is tricky, as the usage above is when a URL is failing to be parsed, so couldn't be redacted this way anyway.

To avoid regressions from this PR, I think it might be enough to tweak the error message above to remove references to the invalid URL and parse error entirely. Signalling that a URL is invalid is likely enough information for somebody to go and find the problem.